tap_mailchimp/sync.py

```diff
@@ -278,8 +278,17 @@ def transform_activities(records):
             file = tar.next()
             while file:
                 if file.isfile():
-                    rawoperations = tar.extractfile(file)
-                    operations = json.loads(rawoperations.read().decode('utf-8'))
+                    try:
+                        rawoperations = tar.extractfile(file)
+                        file_content = rawoperations.read().decode('utf-8')
+                        operations = json.loads(file_content)
+                    except json.JSONDecodeError as e:
+                        if e.args[0] == 'Expecting value: line 1 column 1 (char 0)' and len(file_content.strip()) == 0:
+                            LOGGER.info("Skipping the empty file: %s", file.name)
+                            operations = []
+                        else:
+                            raise Exception("Invalid file format: %s" % file.name) from e
```

Comment:
why do we need to raise an exception if the file with invalid format was getting skipped in the previous code?

Reply:
i take back my comment, thought the else part was aligned with the earlier if statement